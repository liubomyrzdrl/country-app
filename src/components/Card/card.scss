.card {
  display: flex;
  position: relative;
  flex: 0 1 768px;
  background-color: #f1f1f1;
  box-shadow: 0 4px 30px #808080;
  border-radius: 15px;
}

.card-mobile {
  border-radius: 0;
  height: 226px;
  box-shadow: 0px 4px 16px rgba(0, 0, 0, 0.1);
}
