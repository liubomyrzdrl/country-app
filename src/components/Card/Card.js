import PropTypes from 'prop-types';
import React from 'react';
import './card.scss';

export const Card = ({ children, className }) => (
  <div className={`card ${className}`}>
    {children}
  </div>
);

Card.propTypes = {
  children: PropTypes.node,
  className: PropTypes.string,
};
