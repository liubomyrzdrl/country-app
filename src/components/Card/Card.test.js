/* eslint-disable testing-library/no-container */
/* eslint-disable testing-library/no-node-access */
import React from 'react';
import { render, screen } from '@testing-library/react';
import { Card } from './Card';

describe('Test Card', () => {
  it('should render Card component without error', () => {
    render(<Card />);
  });

  it('should render Card component element by class "card"', () => {
    const { container } = render(<Card />);
    expect(container.getElementsByClassName('card')[0]).toBeInTheDocument();
  });

  it('should render Card component element by text child "Test"', () => {
    render(<Card>Test</Card>);
    expect(screen.getByText('Test')).toBeInTheDocument();
  });

  it('should render Card component element by props className - "card-mobile"', () => {
    const { container } = render(<Card className="card-mobile" />);
    expect(container.getElementsByClassName('card-mobile')[0]).toBeInTheDocument();
  });
});
