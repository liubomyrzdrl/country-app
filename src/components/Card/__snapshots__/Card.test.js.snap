// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Card should render Card component without error 1`] = `
<div>
  <div
    class="card "
  />
</div>
`;
